### ISSUE TITLE
Unable to add Bold lock to Home Assistant

### ISSUE THREAD

**Issue by jamespreedy** [lwestenberg/homeassistant_bold#25]
Hi folks, thanks for your help in advance for this one.

My Bold device stopped working in Home Assistant today.
I deleted them, deleted the HACS repo, redownloaded it (2ca295d) and tried to re-add my lock.

I complete the code and sign in on the Bold side, but when it passes back to Home Assistant I get the below error.

Oops... an error occured
Unexpected response Too Many Requests

Screenshots below. Is there any further information I can provide that might help with troubleshooting this?

Many thanks,

James.

![Screenshot 2023-07-30 at 7 46 36 pm](https://github.com/lwestenberg/homeassistant_bold/assets/559531/f61bb73b-2492-45f8-9f58-bf709a8f6667)

**Comment by tim427** [comment]
Hi James,

Sounds like some issues at Nabu Casa. Have you tried all the steps on a later time?

**Comment by bartbakels** [comment]
Seems like again an issue with the bold servers, same issue here since today. This happened several times last months

**Comment by titom43** [comment]
Same problem here!
But I don't use Nabu Casa, I'm using a standalone version (accessible from own subdomain).

I was able to setup a previous version...

Best regards,
Thomas

**Comment by zumheulen** [comment]
Hello everyone!  I too have the same problem since yesterday!  All attempts to reinstall it have so far been futile!  I don't use nabucasa but cloudflair if it helps.

**Comment by titom43** [comment]
Alright. Will try later on (still not working)…

In case of, I’m also behind Cloudflare…

**Comment by tim427** [comment]
I've contacted Bold and indeed; they ratelimit their OAuth service (for a good reason!), but all the authentication comes from Nabu Casa (same IP-address, over and over again), and therefor hitting the threshold.

Nabu Casa doesn't have any back-pressuring (yet?), and therefore retrying the rate-limited request, which makes everything worse :)

TLDR: Bold, Nabu Casa and I are currently brainstorming for a short and long term solution.

**Comment by titom43** [comment]
Thank you for the update.

It sounds to me that previous version was Nuba Casa free. Am I wrong?

Would it be possible to have an (advanced user) option to bypass Nuba Casa and get the OAuth done directly by the Internet accessible HA instance?

**Comment by tim427** [comment]
Technically, yes. But then each use has to register its own application with Bold, etc.

Now the App-secret is with Nabu Casa, allowing everyone to authenticate via Nabu Casa, without signing-up with Bold.

BTW: This is completely separate from the paid Nabu Casa service.

**Comment by titom43** [comment]
Alright. I think that I got it.

You mean that now that Bold opened its app to third party app, we have to get the 3rd party registred by Bold to be able to create a link.

Before, it was emulating the app connection and then was not implementing the 3rd party API mechanisms.

Then I have 2 questions:
- having the link with Nabu Casa created means that anyone linked to NC can access to my Bold locks? Or NC is just acting as a proxy and only my HA instance is linked?
- is it difficult to get MyApp registred at Bold or just some HTTP requests?

Thank you for your appreciated support.

**Comment by tim427** [comment]
Thanks, this applies for everyone ;)

**Comment by zumheulen** [comment]
@tim427 Many thanks to you and everyone for your efforts and time invested. I really appreciate that!

Oh, I don't know if that's appropriate now, but the lock also has a battery status, is it planned to show this as an entity in HA?

**Comment by tim427** [comment]
As a temporary solution Bold killed the OAuth for a while, so the rate-limit shrunk; everything should work now.

When I'll have some time, I'll try to add some back-pressuring on this integration.

PS: also the the translation of "battery status" is on my roadmap :)

**Comment by plasterer41** [comment]
Thanks very much for this 👍🏼 working again now for me.

**Comment by zumheulen** [comment]
Thank you very much, the registration also worked for me again without reinstallation.  All door locks can be operated

**Comment by jamespreedy** [comment]
@Golgautier - yes, my door still unlocks.
I've not tried removing and re-adding the integration though I'm afraid.

**Comment by plasterer41** [comment]
Mine are still working. On 7 Aug 2023, at 13:51, Gautier LEBLANC ***@***.***> wrote:﻿
Hi everyone,
Unfortunately, I have the issue. I cannot say if appreared long time ago or these to last days (I have not played with HA for weeks). I juste restarted HA few times today, and it could explain this status...
It is still working fine for you ?

**Comment by tim427** [comment]
> Arg... It is strange...
> 
> Actually I am surprised that all authentication request comes from Nabu ? Why the HA server does not do it by itself ?

I'll qoute myself;

> Now the App-secret is with Nabu Casa, allowing everyone to authenticate via Nabu Casa, without signing-up with Bold.

Otherwise every HA installation requires it's own OAuth App secret, callback URL, etc. etc. -> Bold will *not* provide this, as the administration will be grow exponentially.
Nabu Casa has a OAuth proxy for many years and is exactly solving this issue (and works for a lot of different integrations).
So the only thing we still need to do; a back-pressuring system.

Feel free to create a pull request ;)

**Comment by zumheulen** [comment]
Oh no oh no oh no! This afternoon we found out that the locks cannot be operated again! The error message from nabucasa comes again.
Please don't tell me it's just me :-/

**Comment by plasterer41** [comment]
I have contacted nabu casa and raised the issue. I have updated my review on Amazon to warn other potential home assistant purchasers of these locks. I can’t understand why Bold are not working on a full native implementation. It really is a pain in the ass and I’m tempted to just sell my locks and connects and look for another solution 😩

**Comment by plasterer41** [comment]
When I contacted bold, they simply point you to this custom integration. Not good enough for a company to not actively look to enhance their products.

**Comment by zumheulen** [comment]
As good as the product is, smart doesn't stop there "for me", only with its own app ec.  to run.  Bold should be ashamed not to be actively involved here.  If my family and I didn't find these motorized locks like the Nuci so ugly and noisy, I probably wouldn't have gotten the Bold.  But it's small, rugged, quiet, stylish and we've color matched it by powder coating it.

**Comment by zumheulen** [comment]
Sorry if that came across as blaming you.  I understand that very well and you have the greatest respect for this work, which should actually be Bold's job because this company wants to sell its "smart products".  Who as a provider today the alternative market was ZB.  Homeassistant refuses, I think he will lose the race.  After the chts nevertheless, thank you very much for your work.  I/we will continue to provide feedback here if desired.  Best regards

**Comment by plasterer41** [comment]
Totally agree with you!On 9 Aug 2023, at 09:26, zumheulen ***@***.***> wrote:﻿
Sorry if that came across as blaming you.  I understand that very well and you have the greatest respect for this work, which should actually be Bold's job because this company wants to sell its "smart products".  Who as a provider today the alternative market was ZB.  Homeassistant refuses, I think he will lose the race.  After the chts nevertheless, thank you very much for your work.  I/we will continue to provide feedback here if desired.  Best regards

**Comment by plasterer41** [comment]
Hey, I’m not having a go at anyone on here. Quite the opposite. You guys shouldn’t be solving a gap in Bold’s product. My point is that Bold as a company should be gleaning your expertise to enhance their product in an official way. I for one am very grateful for guys like you 😘Sent from my iPhoneOn 9 Aug 2023, at 08:59, Tim de Boer ***@***.***> wrote:﻿
@plasterer41 and @zumheulen this integration has nothing to do with the company Bold and/or Nabu Casa.
Please understand this plugin/integration is still in an unofficial state (not in HomeAssistant yet, but via HACS) and currently only developed by me and @lwestenberg
Both of us have full time dayjobs, etc. If you want to speed things up; feel free to contribute by creating a pull request.
Otherwise; please be patient, we are working on it (in our free time).
If this problem is fixed (and some extra features are added), we are planning to make it an official integration within HomeAssistant.

**Comment by jamespreedy** [comment]
Hey, as we're all on the feedback train I'd like to jump in.

I really appreciate the work you folks do with this integration - its never easy to find the time for this stuff, and I wish I had a big brain to contribute in a meaningful way.

I found it a little distasteful that Bold claim they have Home Assistant support, and then push their customers to a third party community supported integration - thats not fair on anyone.

Their support teams had been telling me for a year or so that HomeKit support was in their pipeline, and then out of the blue they recently said it no longer was. Shame as there aren't many similar lock alternatives for now (euro cylinder + no motorised components).

I feel they really need to step up. Either add HomeKit support or a 1st party integration. If not, support you folks and Nabu Casa properly when they are issues.

I'd only recently replaced my HomeBridge install (which I used to pull back into Home Assistant) with this integration - I'll switch back for now until this is back up and running. (HomeBridge seems to be working ok for me at the moment.)

Anyway. Thank you, thank you, thank you. We all appreciate it.
You folks make Home Assistant an awesome place to be. 🍺

**Comment by A138SFM** [comment]
> Hey, as we're all on the feedback train I'd like to jump in.
> 
> I really appreciate the work you folks do with this integration - its never easy to find the time for this stuff, and I wish I had a big brain to contribute in a meaningful way.
> 
> I found it a little distasteful that Bold claim they have Home Assistant support, and then push their customers to a third party community supported integration - thats not fair on anyone.
> 
> Their support teams had been telling me for a year or so that HomeKit support was in their pipeline, and then out of the blue they recently said it no longer was. Shame as there aren't many similar lock alternatives for now (euro cylinder + no motorised components).
> 
> I feel they really need to step up. Either add HomeKit support or a 1st party integration. If not, support you folks and Nabu Casa properly when they are issues.
> 
> I'd only recently replaced my HomeBridge install (which I used to pull back into Home Assistant) with this integration - I'll switch back for now until this is back up and running. (HomeBridge seems to be working ok for me at the moment.)
> 
> Anyway. Thank you, thank you, thank you. We all appreciate it. You folks make Home Assistant an awesome place to be. 🍺

There is a home kit solutions and I had it running for a couple of years https://github.com/StefanNienhuis/homebridge-bold#readme

**Comment by jamespreedy** [comment]
Yep, thats the one I'm using - seems to work OK.

When I said they were doing a HomeKit solution, Bold had told me that they were adding native HK support to the Bold Connect Hub. That would remove the need for an intermediary solution like Bold => HomeBridge => Home Assistant via HK.

**Comment by ChristianPi99** [comment]
Does the Bold integration with Home Assistant still work or doesn't it work anymore?  Based on the above I am not really sure, if only some users are having problems with it or if it doesn't work at the moment and I should consider the HomeBridge integration?

**Comment by plasterer41** [comment]
Works perfect for me also. Sent from my iPhoneOn 4 Apr 2024, at 15:35, ChristianPi99 ***@***.***> wrote:﻿
Does the Bold integration with Home Assistant still work or doesn't it work anymore?  Based on the above I am not really sure, if only some users are having problems with it or if it doesn't work at the moment and I should consider the HomeBridge integration?

**Comment by GJCD** [comment]
I don't know what I did but now everytime I restart HomeAssistant, the Bold Integration doesn't load and I have to add it again.
I tried to delete it in custum_components and reinstall it but nothing works.
If anyone can help I would appreciate.
Thank you

**Comment by Chickens66** [comment]
Hi, I’m seeing what looks like the same issue again with Bold 0.1.4.

The integration worked normally until around August 12, 2026. Since then all Bold entities have been unavailable.

I have already:

* disabled the integration and waited
* restarted Home Assistant
* enabled it again
* tried a single reload
* enabled debug logging and reproduced the error

The debug log consistently shows:

429 Too Many Requests

for:

https://account-link.nabucasa.com/refresh_token/bold

The relevant traceback ends during the OAuth token refresh, before the Bold entities can be loaded.

I also updated Home Assistant Core today, but the Bold integration had already been unavailable for several days before that update, so the Core update does not appear to be the cause.

Current setup:

* Bold custom integration: 0.1.4
* 3 Bold devices / 3 entities
* Home Assistant Cloud account linking
* Issue started approximately August 12, 2026
* Still reproducible on August 16, 2026

Reloading, restarting Home Assistant, and waiting for the rate limit to clear have not resolved it.

Is the shared Home Assistant/Nabu Casa OAuth client for Bold currently being rate-limited, or has anything changed on the Bold OAuth side?

I can provide the complete debug log if helpful.
